oxidation/libparsec/crates/types/src/shamir.rs

```diff
@@ -100,3 +108,46 @@ impl_transparent_data_format_conversion!(
     reveal_token_share,
     data_key_share,
 );
+
+impl ShamirRecoveryShareData {
+    pub fn dump(&self) -> Result<Vec<u8>, &'static str> {
+        ::rmp_serde::to_vec_named(self).map_err(|_| "Serialization failed")
+    }
+
+    pub fn load(buf: &[u8]) -> Result<Self, &'static str> {
+        ::rmp_serde::from_slice(buf).map_err(|_| "Deserialization failed")
+    }
+
+    pub fn decrypt_verify_and_load_for(
+        ciphered: &[u8],
+        recipient_privkey: &PrivateKey,
+        author_verify_key: &VerifyKey,
+    ) -> Result<ShamirRecoveryShareData, &'static str> {
```

Comment:
Use an enum instead of a raw string, it simplify testing

Reply:
a865d44fa